erroranalysis/erroranalysis/_internal/cohort_filter.py

```diff
@@ -279,18 +279,21 @@ def _add_filter_cols(self, df, filters):
         else:
             df[TRUE_Y] = self.true_y
 
-        if self.model is None:
-            df[PRED_Y] = self.pred_y
-
         if not is_spark(df):
             df[ROW_INDEX] = np.arange(0, len(self.true_y))
+
+        if self.pred_y is not None:
+            df[PRED_Y] = self.pred_y
+        else:
+            if not is_spark(self.dataset):
+                if not isinstance(self.dataset, pd.DataFrame):
+                    df[PRED_Y] = self.model.predict(self.dataset)
```

Comment:
Here we may be calculating PRED_Y even if we don't need it for filtering?
That is why the calculation was done in has_classification_outcome instead previously, so we would only call predict (which is expensive) if we really need it:

```
        if has_classification_outcome:
            if PRED_Y in df:
                pred_y = df[PRED_Y]
            else:
                # calculate directly via prediction on model
                pred_y = self.model.predict(
                    df.drop(columns=[TRUE_Y, ROW_INDEX]))
```

Reply:
We can just have a similar check to has_classification_outcome, such as ```has_predicted_Y```, if there is a "Predicted Y" filter passed here - and only then call the predict function.